### PR TITLE
Add some tests and fix some exceptions during the tests

### PR DESCRIPTION
**Pull request by xconverge**
Cant run them all on my system, want to see if travis can...

Yay! We love PRs! 🎊

Please include a description of your change and ensure:

- [X] Commit message has a short title & issue references
- [X] Each commit does a logical chunk of work.
- [X] It builds and tests pass (e.g `gulp`)

More info can be found on our [contribution guide](https://github.com/VSCodeVim/Vim/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by johnfn** [comment]
I'm all for new tests!

**Comment by johnfn** [comment]
Oops, I had some pending comments here that didn't get added, I think...

File: test/mode/modeNormal.test.ts [hunk 1]
```diff
@@ -933,6 +933,20 @@ suite("Mode Normal", () => {
     });
 
     newTest({
+      title: "Can handle 'J' followed by x",
```
Comment (xconverge) [inline test/mode/modeNormal.test.ts]: @rebornix this one is duplicated now right?

Comment (rebornix) [inline test/mode/modeNormal.test.ts]: sorry it is :)

File: test/mode/modeInsert.test.ts [hunk 2]
```diff
@@ -43,7 +43,18 @@ suite("Mode Insert", () => {
         assertEqual(TextEditor.getSelection().start.character, 4, "<Esc> moved cursor position.");
     });
 
-    test("", async () => {
+   test("<C-c> can exit insert", async () => {
```
Comment (rebornix) [inline test/mode/modeInsert.test.ts]: I think it's `useCtrlKeys`

Comment (xconverge) [inline test/mode/modeInsert.test.ts]: We have <C- > keys all throughout our tests though

**Pull request merged**